src/main/java/seedu/clinic/model/supplier/SupplierPredicate.java

```diff
@@ -0,0 +1,84 @@
+package seedu.clinic.model.supplier;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import seedu.clinic.commons.util.StringUtil;
+import seedu.clinic.model.product.Product;
+
+/**
+ * Tests whether the {@code Name}, {@code Remark} or any of the {@code Product} sold by {@code Supplier} matches any
+ * of the keywords given.
+ */
+public class SupplierPredicate implements Predicate<Supplier> {
+    private final Optional<List<String>> nameKeywords;
+    private final Optional<List<String>> productKeywords;
+    private final Optional<List<String>> remarkKeywords;
+
+    /**
+     * Constructs a new {@code SupplierPredicate}.
+     */
+    public SupplierPredicate(List<String> nameKeywords, List<String> productKeywords,
```

Comment:
Not sure if it's helpful but one way to make it more concise is to initialize all nameKeywords, productKeywords and remarkKeywords to Optional.empty() then only keep 3 if statement to update them if necessary